### PR TITLE
Reversing every word in a String

### LINKED ISSUE #1921: C/CPP Reversing a word in a string
### Description

Documentation on how reverse a word  in a string

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by yellowberard**
<hr>

## Description 📜
To reverse every word in a string.
Fixes #1921

<hr>

## Type of change 📝

- [X] Doc#umentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [X] C/#CPP
<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Comment by yellowberard** [comment]
@kashishahuja2002 Please review my PR

**Comment by yellowberard** [comment]
closes pr #5665

**Review by vasu-1 — CHANGES_REQUESTED** [review]
- First of all add your file link into respective folder's readme.md
- remove unneccessary empty lines
- beautify your document with markdown formatting
- add space complexities

**Comment by yellowberard** [comment]
I have taken the problem statement and examples from leetcode so do I have to change the language of the question. @vasu-1

**Comment by vasu-1** [comment]
> 

Please check it once , i have updated the requested changes

**Comment by yellowberard** [comment]
@vasu-1 I have made the requested changes.

File: C_CPP/Competitive Programming/Reversing_every_word_in_string.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+### Time Complexity
+
+O(n)
+
+### Space Complexity
+
+O(1)
+
+#### Author
+- [Mayank Goel](https://github.com/yellowberard) For documentation
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Reversing_every_word_in_string.md:56]: No need
Please REmove

File: C_CPP/Competitive Programming/Reversing_every_word_in_string.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+while(w>>word)
+{
+    reverse(word.begin(),word.end());  //reversing the word so that it comes to original form
+    rev_word += word;     //storing word
+    rev_word += " ";     //adding space
+}
+
+rev_word.erase(rev_word.length()-1,1); //removing the additional last space added
+return rev_word;
+}
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Reversing_every_word_in_string.md:43]: Put whole code !

**Review by vasu-1 — CHANGES_REQUESTED** [review]

**Comment by yellowberard** [comment]
@vasu-1 I have made the requested changes.

File: C_CPP/Competitive Programming/Reversing_every_word_in_string.md [hunk 3]
```diff
@@ -23,25 +23,37 @@ I have commented the code for explanation of the code and logic.
 ## Solution
 
 ```
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Reversing_every_word_in_string.md:25]:
```suggestion
```cpp
```

for syntax effect

**Review by vasu-1 — CHANGES_REQUESTED** [review]

**Comment by yellowberard** [comment]
@vasu-1 I have made the requested changes.

**Review by vasu-1 — APPROVED** [review]
LGTM @yellowberard 

@Suvraneel @Saviour1001

**Pull request merged**